### ISSUE TITLE
COE panics with CachedOnDiskCorpus

### ISSUE THREAD

**Issue by andreafioraldi** [AFLplusplus/LibAFL#592]
```
thread '<unnamed>' panicked at 'already mutably borrowed: BorrowError', /home/andrea/Desktop/libafl_fuzzbench/LibAFL/libafl/src/corpus/cached.rs:67:21

.....

  17: <libafl::corpus::cached::CachedOnDiskCorpus<I> as libafl::corpus::Corpus<I>>::get
             at /home/andrea/Desktop/libafl_fuzzbench/LibAFL/libafl/src/corpus/cached.rs:67:12
  18: <libafl::schedulers::testcase_score::CorpusPowerTestcaseScore<I,S> as libafl::schedulers::testcase_score::TestcaseScore<I,S>>::compute
             at /home/andrea/Desktop/libafl_fuzzbench/LibAFL/libafl/src/schedulers/testcase_score.rs:89:36
  19: <libafl::stages::power::PowerMutationalStage<E,F,EM,I,M,O,OT,S,Z> as libafl::stages::mutational::MutationalStage<E,EM,I,M,S,Z>>::iterations
             at /home/andrea/Desktop/libafl_fuzzbench/LibAFL/libafl/src/stages/power.rs:70:21
  20: <libafl::stages::power::PowerMutationalStage<E,F,EM,I,M,O,OT,S,Z> as libafl::stages::mutational::MutationalStage<E,EM,I,M,S,Z>>::perform_mutational
             at /home/andrea/Desktop/libafl_fuzzbench/LibAFL/libafl/src/stages/power.rs:95:19
  21: <libafl::stages::power::PowerMutationalStage<E,F,EM,I,M,O,OT,S,Z> as libafl::stages::Stage<E,EM,S,Z>>::perform
             at /home/andrea/Desktop/libafl_fuzzbench/LibAFL/libafl/src/stages/power.rs:167:19
  22: <(Head,Tail) as libafl::stages::StagesTuple<E,EM,S,Z>>::perform_all
             at /home/andrea/Desktop/libafl_fuzzbench/LibAFL/libafl/src/stages/mod.rs:109:9
  23: <(Head,Tail) as libafl::stages::StagesTuple<E,EM,S,Z>>::perform_all
             at /home/andrea/Desktop/libafl_fuzzbench/LibAFL/libafl/src/stages/mod.rs:113:9
  24: <libafl::fuzzer::StdFuzzer<CS,F,I,OF,OT,S> as libafl::fuzzer::Fuzzer<E,EM,I,S,ST>>::fuzz_one
             at /home/andrea/Desktop/libafl_fuzzbench/LibAFL/libafl/src/fuzzer/mod.rs:544:9
```

likely, it is trying to borrow the current testcase

**Comment by tokatoka** [comment]
I fixed this in the past https://github.com/AFLplusplus/LibAFL/pull/295
but it seems I made the same mistake again during power schedule refactoring

**Comment by tokatoka** [comment]
fixed in #593
